src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -140,7 +153,23 @@ public void run() {
                         LOGGER.warn("Failed to parse the polling log", e);
                         cause = new GitHubPushCause(pushBy);
                     }
-                    if (asParameterizedJobMixIn(job).scheduleBuild(cause)) {
+
+                    // Get the parameters defined by the job, as we will be overriding certain values
+                    List<ParameterValue> values = getDefaultParameters();
+                    Iterator<ParameterValue> it = values.iterator();
+                    while (it.hasNext()) {
+                        ParameterValue pv = it.next();
+                        if (parameterIsOverridable(pv.getName())) {
+                            it.remove();
+                        }
+                    }
+                    values.add(new StringParameterValue("ref", ref));
+                    values.add(new StringParameterValue("head", head));
+
+                    // Schedule the job with the parameters and cause
+                    QueueTaskFuture<?> build = asParameterizedJobMixIn(job)
+                            .scheduleBuild2(0, new ParametersAction(values), new CauseAction(cause));
```

Comment:
Not entirely surely what you mean, do you mean add the parameters as environment variables only?  Or add as environment variables as well as job parameters similar to what the Jenkins Github Pull Request Builder plugin does?